app/policies/application_policy.rb

```diff
@@ -13,10 +13,14 @@ def admin_user?
   end
 
   def regular_user?
-    @user.is_a?(Person)
+    @user.is_a?(Person) && @user.third_party_user? == false
   end
 
   def readonly_user?
-    @user.is_a?(ReadonlyUser)
+    @user.is_a?(ReadonlyUser) or @user.third_party_user? && @user.super_admin? == false
```

Comment:
Rather than changing these definitions, I would suggest that when you load a SSCL user, you set them as a ReadonlyUser by making a change to `Login.current_user`